crates/astria-conductor/src/data_availability/sync.rs

```diff
@@ -0,0 +1,121 @@
+use celestia_client::{
+    celestia_types::{
+        nmt::Namespace,
+        Height,
+    },
+    jsonrpsee::http_client::HttpClient,
+    CelestiaClientExt as _,
+    SEQUENCER_NAMESPACE,
+};
+use color_eyre::eyre::{
+    self,
+    WrapErr as _,
+};
+use futures::stream::FuturesOrdered;
+use tokio::select;
+use tracing::{
+    info,
+    instrument,
+    warn,
+};
+
+use crate::{
+    block_verifier::BlockVerifier,
+    data_availability::{
+        send_sequencer_subsets,
+        verify_sequencer_blobs_and_assemble_rollups,
+        SequencerBlockSubset,
+    },
+    executor,
+};
+
+#[instrument(name = "sync DA", skip_all)]
+pub(crate) async fn run(
+    start_sync_height: u32,
+    end_sync_height: u32,
+    namespace: Namespace,
+    executor_tx: executor::Sender,
+    client: HttpClient,
+    block_verifier: BlockVerifier,
+) -> eyre::Result<()> {
+    use futures::{
+        FutureExt as _,
+        StreamExt as _,
+    };
+
+    let mut height_stream = futures::stream::iter(start_sync_height..end_sync_height);
+    let mut block_stream = FuturesOrdered::new();
+
+    'sync: loop {
+        let client = client.clone();
+        let block_verifier = block_verifier.clone();
```

Comment:
I believe this is because of the `async move`s in the `select!`. 
When those let statements are removed, you get a `use of moved value` error. Should the `get_new_sequencer_block_data_with_retry` and `verify_sequencer_blobs_and_assemble_rollups` functions be changed to take the `BlockVerifyer` and the `HttpClient` as references? Or is there a specific reason not to do that?

Reply:
clone these objects right where you need them and then `async move` them into the blocks.


Also, the async blocks I am seeing also do yet another `clone` on the verifier and client, which is redundant